src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -99,22 +99,69 @@ internal override bool IsInsertionTrigger(SourceText text, int characterPosition
             return ValueTuple.Create(displayText, insertionText);
         }
 
-        private static CompletionItemRules s_importDirectiveRules =
-            CompletionItemRules.Create(commitCharacterRules: ImmutableArray.Create(CharacterSetModificationRule.Create(CharacterSetModificationKind.Replace, '.', ';')));
-        private static CompletionItemRules s_importDirectiveRules_preselect = s_importDirectiveRules.WithSelectionBehavior(CompletionItemSelectionBehavior.HardSelection);
+        protected override CompletionItemRules GetCompletionItemRules(List<ISymbol> symbols, SyntaxContext context, bool preselect)
+        {
+            return cachedRules[Tuple.Create(context.IsInImportsDirective, preselect,
+                ((CSharpSyntaxContext)context).IsPossibleTupleContext)] ??
```

Comment:
Only trouble is that those rules are not yet implemented for VB. I'll just pass false for now.